### PR TITLE
Update Docs for AWS SVIDStore plugin to include required IAM permissions

### PR DESCRIPTION
**Pull request by davidgarc**
**Pull Request check list**

- [x ] Commit conforms to CONTRIBUTING.md?
- [ N/A] Proper tests/regressions included?
- [ x] Documentation updated?

**Affected functionality**
None

**Description of change**
Updated documentation of the SVIDStore AWS plugin to include required AWS IAM permissions for it to work properly.

**Which issue this PR fixes**
None

### CODE REVIEW

File: doc/plugin_agent_svidstore_aws_secretsmanager.md [hunk 1]
```diff
@@ -18,6 +18,49 @@ The format that is used to store in a secret the issued identity is the followin
 }
 ```
 
+### Required AWS IAM permissions
```
Comment (MarcosDY) [inline doc/plugin_agent_svidstore_aws_secretsmanager.md:21]: what do you think about follow the same structure used on [gcp plugin](https://github.com/spiffe/spire/blob/main/doc/plugin_agent_svidstore_gcp_secretmanager.md#required-gcp-permissions)?  

And maybe avoid adding an example, so we don't have to worry about keeping it updated?

Comment (davidgarc) [reply doc/plugin_agent_svidstore_aws_secretsmanager.md:21]: Yes good point, I'll remove the example.

File: doc/plugin_agent_svidstore_aws_secretsmanager.md [hunk 2]
```diff
@@ -18,6 +18,21 @@ The format that is used to store in a secret the issued identity is the followin
...
+```
+secretsmanager:DescribeSecret
+secretsmanager:CreateSecret
+secretsmanager:RestoreSecret
+secretsmanager:PutSecretValue
+secretsmanager:DeleteSecret
+kms:Encrypt
+```
+
+Please note that this plugin does not require permission to read secret payloads stored.
```
Comment (azdagron) [inline doc/plugin_agent_svidstore_aws_secretsmanager.md:34]: Just a small suggestion to clarify this wording. The way it was phrased could be misinterpreted to imply that the plugin reads the secrets but doesn't need perms to do so.
```suggestion
Please note that this plugin does not read secrets it has stored and therefore does not require read permissions.
```

**Review by azdagron — DISMISSED** [review]

**Comment by azdagron** [comment]
Thanks, @davidgarc !

**Comment by davidgarc** [comment]
> Thanks, @davidgarc !

sorry was trying to fix the DCO signoff but ended up rebasing and pulling a bunch unrelated commits to this changes. Working on fixing this.

**Review by azdagron — APPROVED** [review]
No worries :) Thanks for fixing it up.

**Pull request merged**